new-integration-tests/src/test/java/oracle/weblogic/kubernetes/extensions/IntegrationTestWatcher.java

```diff
@@ -85,21 +89,20 @@ public boolean supportsParameter(ParameterContext parameterContext,
   @Override
   public Object resolveParameter(ParameterContext parameterContext,
       ExtensionContext extensionContext) throws ParameterResolutionException {
-    Object requiredTestInstance = extensionContext.getRequiredTestInstance();
-    List<String> namespaces = null;
-
-    if (requiredTestInstance.getClass().isAnnotationPresent(ITNamespaces.class)) {
-      ITNamespaces itNsTag = requiredTestInstance.getClass().getAnnotation(ITNamespaces.class);
-      logger.info("Creating {0} unique namespaces for the test", itNsTag.numofns());
-      namespaces = new ArrayList();
-      for (int i = 1; i <= itNsTag.numofns(); i++) {
-        String namespace = assertDoesNotThrow(() -> createUniqueNamespace(),
-            "Failed to create unique namespace due to ApiException");
-        namespaces.add(namespace);
-        logger.info("Created a new namespace called {0}", namespace);
-      }
+    Namespaces ns = parameterContext.findAnnotation(Namespaces.class).get();
+    List<String> namespaces = new ArrayList();
+    for (int i = 1; i <= ns.value(); i++) {
```

Comment:
use for over collection

Reply:
Sorry I don't understand what you mean by "use for over collection" , can you elaborate ?

Reply:
Looking at this again, @markxnelson, he's not iterating over a collection but generating the required count of namespaces per the value from the annotation.

Reply:
ah ok, i see, thanks, please ignore